### PR TITLE
Initial Deployment

### PR DESCRIPTION
**Pull request by stevenwjy**
Available features:
- Main layout (header, footer)
- Welcome page (banner)
- Past NUANSA (overview)
- User login (access blocked in production)

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.present?
```
Comment (donjar) [inline app/models/ability.rb]:
```suggestion
    return unless user.present?
```

Suggest using Rubocop :)

File: app/models/role.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Role < ApplicationRecord
+has_and_belongs_to_many :users, :join_table => :users_roles
```
Comment (donjar) [inline app/models/role.rb]: Indentation

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
   <head>
-    <title>Nuansacp</title>
+    <title>NUANSA</title>
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+
+    <%= stylesheet_link_tag    'font-awesome/css/font-awesome.min.css', 'data-turbolinks-track': 'reload' %>
```
Comment (donjar) [inline app/views/layouts/application.html.erb:11]: Kaya gini gabung ama asset pipeline ngga ya? Lupa. Kalo ini ditaro di folder `vendor/` harusnya bisa gabung ama asset pipeline sih.

File: app/views/past_events/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,126 @@
+<div class="timeline">
+	<ul>
```
Comment (donjar) [inline app/views/past_events/index.html.erb:2]: 2 spaces :)

File: app/views/past_events/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+		<li>
+			<div class="content">
+				<h3>Tasasek: The Courage To Choose</h3>
+				<img src="../assets/2018logo.jpg" class="logo" alt="Tasasek">
```
Comment (donjar) [inline app/views/past_events/index.html.erb]: Use `image_tag`

File: config/initializers/assets.rb [hunk 6]
```diff
@@ -11,4 +11,36 @@
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in the app/assets
 # folder are already added.
-# Rails.application.config.assets.precompile += %w( admin.js admin.css )
+Rails.application.config.assets.precompile += %w(
```
Comment (donjar) [inline config/initializers/assets.rb:14]: I'm not sure, ini bakal masukin semuanya ke asset pipeline yah?

**Review by donjar — COMMENTED** [review]
Generally looks great!

- gw ga yakin kalo Rails yang skarang asset pipelinenya kaya gimana. Make sure kalo app nya in production mode, CSS / Javascript file nya 1 doang.
- Not sure kalo file2 CSS/JS nya taro di lib. Setau gw 3rd party code lebih cocok di vendor.
- Do set up linters supaya code style nya bisa set in stone :)

File: app/models/ability.rb (hunk 1 above)
Comment (stevenwjy) [reply app/models/ability.rb]: @Yehezkiel01

File: app/views/past_events/index.html.erb (hunk 5 above)
Comment (stevenwjy) [reply app/views/past_events/index.html.erb]: Yes, I have updated it on the other branch, will be merged to development branch soon.

File: app/views/past_events/index.html.erb (hunk 4 above)
Comment (stevenwjy) [reply app/views/past_events/index.html.erb:2]: Ok, noted

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (stevenwjy) [reply app/views/layouts/application.html.erb:11]: Gabung sih, yang di vendor/ itu dari yg template dipake, yg di lib/ itu yg emg pure CSS library

File: app/models/role.rb (hunk 2 above)
Comment (stevenwjy) [reply app/models/role.rb]: Ok, noted.

File: app/models/ability.rb (hunk 1 above)
Comment (stevenwjy) [reply app/models/ability.rb]: Btw, karena cm butuh minor change, I will add this change di next PR skalian fix bbrp minor changes lain ya buat code styling.

File: config/initializers/assets.rb (hunk 6 above)
Comment (stevenwjy) [reply config/initializers/assets.rb:14]: Yup

**Pull request merged**